posthog/models/test/test_person_override_model.py

```diff
@@ -9,392 +9,580 @@
 
-    old_mapping = PersonOverrideMapping.objects.create(
-        team_id=team.id,
-        uuid=old_person_id,
-    )
-    override_mapping = PersonOverrideMapping.objects.create(
-        team_id=team.id,
-        uuid=override_person_id,
-    )
+            created.append(person_override)
 
-    p1 = PersonOverride.objects.create(
-        team=team,
-        old_person_id=old_mapping,
-        override_person_id=override_mapping,
-        oldest_event=oldest_event,
-        version=1,
-    )
-    p1.save()
+        assert all(p.override_person_id == override_mapping for p in created)
+        assert len(set(p.old_person_id.uuid for p in created)) == n_person_overrides
 
-    assert p1.old_person_id == old_mapping
-    assert p1.override_person_id == override_mapping
+    """
+    Concurrency tests for person overrides table
+    Goal: verify that we don't end up in a situation with the same uuid is both
+    an old person id and an override person id
+    - there are two cases that we want to check for
+        - concurrent merges
+        - concurrent merge and person deletion
+    In both cases one of the transactions will wait on the lock,
+    so they can only complete in one order (which is tested below).
+    Note that to test the race condition scenario we need to:
+        1. create multiple concurrent transactions, such that we can verify
+        constraints are enforced at COMMIT time.
+        2. enable transactions for the Django test. This is more so we can see data
+        from the main Django PostgreSQL connection session in the other
+        concurrent transactions. Not 100% required but makes things a little
+        easier to write.
+    """
 
-    new_old_mapping = PersonOverrideMapping.objects.create(
-        team_id=new_team.id,
-        uuid=new_old_person_id,
-    )
-    new_override_mapping = PersonOverrideMapping.objects.create(
-        team_id=new_team.id,
-        uuid=old_mapping.uuid,
-    )
+    # def test_person_override_merge(self, people, team, oldest_event):
+    def test_person_override_merge(self):
+        """Verify our merge function work as expected."""
+        old_person, override_person, _ = self.p1, self.p2, self.p3
 
-    p2 = PersonOverride.objects.create(
-        team=new_team,
-        old_person_id=new_old_mapping,
-        override_person_id=new_override_mapping,
-        oldest_event=oldest_event,
-        version=1,
-    )
-    p2.save()
+        with create_connection() as merge_cursor:
+            merge_cursor.execute("BEGIN")
+            _merge_people(self.team, merge_cursor, old_person.uuid, override_person.uuid, self.oldest_event)
+            merge_cursor.execute("COMMIT")
 
-    assert p1.old_person_id.uuid == p2.override_person_id.uuid
-    assert p1.old_person_id.team_id == p1.override_person_id.team_id
-    assert p2.old_person_id == new_old_mapping
-    assert p1.team != p2.team
+        assert [_[0] for _ in PersonOverrideMapping.objects.all().values_list("uuid")] == [
```

Comment:
I prefer the assertion like this... Have we lost the team equality/not-equality assertion?